torchvision/csrc/deform_conv2d.h

```diff
@@ -11,7 +11,7 @@
 #include "hip/vision_cuda.h"
 #endif
 
-// TODO: put this stuff in torchvision namespace
+namespace {
```

Comment:
Again anonymous namespace for internal methods and classes.

Reply:
Generally anonymous namespaces in header files are a bad idea. They won't prevent people from accessing the declarations and each cpp file that includes the header will end up having its own instantiation of the symbols.

Reply:
The header files should not be having so much code TBH. Fortunately in these files we don't run the risk of conflicts (across devices or across ops), like we do on the actual implementations, so separating the methods is optional.

The reason why we separate it here is to indicate that these are not part of the public API. Do you have an alternative on how to achieve this?

Reply:
That's true. All those functions in the headers are not even marked `inline`. I think the solutions would be to either move the implementations in cpp files, or use a nested namespace like `vision::internal` or `vision::detail`.
I think the latter solution is simpler and we can still change later :)
As a side note, why do you not consider `deform_conv2d` part of the public c++ API?

Reply:
I would be happy to use a more C++-adapted way of doing this.

When I implemented the first version of this, I did it this way to avoid having to write one more header file for every cpp file, and the amount of code in each file was fairly small, but things have changed since then, so it might be worth revisiting that decision in light of the current state of the codebase

Reply:
I think the best way to fix this is move the C++ implementation code into cpp files. Though this means, we will be losing the git blame history. Are we OK with this?

Reply:
Don't put anonymous namespaces in headers. If you need header accessible but private implementations, put it in `namespace detail`. But yes, if you don't actually need the implementation in a header, best to put it in C++. Don't worry too much about blame, it doesn't work half the time anyway (pickaxe is your friend).

Reply:
Sounds good, Ed. I think moving the code out of headers is the best solution for the long term.